packages/evm/src/evm.ts

```diff
@@ -673,7 +699,13 @@ export class EVM implements EVMInterface {
       versionedHashes: message.versionedHashes ?? [],
     }
 
-    const interpreter = new Interpreter(this, this.eei, env, message.gasLimit)
+    const interpreter = new Interpreter(
+      this,
+      this.stateManager,
```

Comment:
Is it really necessary to pass `this.stateManager` and `this.blockchain` in this constructor since they are both properties of `this` (i.e. the EVM object)?

Reply:
Interpreter does currently not have access to EVM currently IIRC (am on mobile)

Reply:
it seems to have as far the the Interpreter constructor goes, but the question is will there be a usecase where to pass a different statemanager than the evm's?

Reply:
Well, using EthersStateManager for instance :)

Reply:
I think you are missing the point here, the question was not about using another StateManager but a different one for EVM and StateManager, so if we need to pass both „this“ and „this.StateManager“, where I would also think the answer is clearly: not.

(„This“ is the EVM, just not to overlook)

I think for this.blockchain question goes as well.

Reply:
I don't think I understand the question. The passed StateManager is part of EVM constructor, and Interpreter also needs access to it?

Reply:
Yes, but we do not need to pass this extra on since we already pass `this` (the EVM) so we can do `evm.stateManager` or `evm.blockchain` inside `Interpreter` and so do not need three Interpreter constructor paramters for this, that's is the whole point here. 🙂